core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCnaDiscreteLongData.java

```diff
@@ -0,0 +1,340 @@
+            && geneticProfile.showProfileInAnalysisTab();
+
+        if (isDiscretizedCnaProfile) {
+            for (CnaEvent.Event event : DaoCnaEvent.getAllCnaEvents()) {
+                existingCnaEvents.put(event, event);
+            }
+            MySQLbulkLoader.bulkLoadOn();
+        }
+
+        CnaImportData toImport = new CnaImportData();
+
+        while ((line = buf.readLine()) != null) {
+            lineIndex++;
+            ProgressMonitor.incrementCurValue();
+            ConsoleUtil.showProgress();
+            this.extractDataToImport(geneticProfile, line, lineIndex, toImport);
+        }
+
+        DaoGeneticProfileSamples.addGeneticProfileSamples(
+            geneticProfileId,
+            newArrayList(toImport.eventsTable.columnKeySet())
+        );
+
+        for (Long entrezId : toImport.eventsTable.rowKeySet()) {
+            boolean added = storeGeneticAlterations(toImport, entrezId);
+            if (added) {
+                storeCnaEvents(toImport, entrezId);
+            } else {
+                ProgressMonitor.logWarning("Values not added to gene with entrezId: " + entrezId + ". Skip creation of cna events.");
+            }
+        }
+
+        ProgressMonitor.setCurrentMessage(" --> total number of samples skipped (normal samples): " + sampleFinder.getSamplesSkipped()
+        );
+        buf.close();
+        MySQLbulkLoader.flushAll();
+    }
+
+    public void extractDataToImport(
+        GeneticProfile geneticProfile,
+        String line,
+        int lineIndex,
+        CnaImportData importContainer
+    ) throws Exception {
+        boolean hasData = !line.startsWith("#") && line.trim().length() > 0;
+        if (!hasData) {
+            return;
+        }
+        String[] lineParts = line.split("\t", -1);
+        CanonicalGene gene = this.getGene(cnaUtil.getEntrezSymbol(lineParts), lineParts, cnaUtil);
+        importContainer.genes.add(gene);
+
+        if (gene == null) {
+            return;
+        }
+
+        int cancerStudyId = geneticProfile.getCancerStudyId();
+
+        String sampleIdStr = cnaUtil.getSampleIdStr(lineParts);
+        Sample sample = sampleFinder.findSample(sampleIdStr, cancerStudyId);
+        sampleProfileImporter.createSampleProfile(sample);
+
+        long entrezId = gene.getEntrezGeneId();
+        int sampleId = sample.getInternalId();
+        CnaEventImportData eventContainer = new CnaEventImportData();
+        Table<Long, Integer, CnaEventImportData> geneBySampleEventTable = importContainer.eventsTable;
+
+        if (!geneBySampleEventTable.contains(entrezId, sample.getInternalId())) {
+            geneBySampleEventTable.put(entrezId, sampleId, eventContainer);
+        } else {
+            ProgressMonitor.logWarning(format("Skipping line %d with duplicate gene %d and sample %d", lineIndex, entrezId, sampleId));
+        }
+
+        eventContainer.geneticEvent = cnaUtil.createEvent(geneticProfile, sample.getInternalId(), lineParts);
```

Comment:
Because of the new long format we only have one cna event per row (instead of all events of a single gene on one row). This would result in many more db queries, so @pvannierop and I decided to 1) collect all events, and 2) store the events in a single query . 

See also: `storeGeneticAlterations` and `storeCnaEvents`.

I'll also add some comments to explain this in the code.